examples/init-function/init_function.bal

```diff
@@ -0,0 +1,16 @@
+// Uninitialized integer variable `value`.
+int value;
+
+// Uninitialized final string variable `name`.
+final string name;
+
+function init() returns error? {
+    // Initialize the variable `value` to 5.
+    value = 5;
+    // Initialize the final variable greeting to `James`.
+    name = "James";
+    
+    if value > 3 {
+        return error(string `${name}: The value should be less than 3, but it set to ${value}`);
```

Comment:
Let's not use string template, and make the error message to a simple string.  Also add a comment, explaining the behavior what happens when error is returned.